apps/emqx_management/src/emqx_mgmt_data_backup.erl

```diff
@@ -871,8 +872,14 @@ backup_dir_old_version() ->
     emqx:get_env(data_dir).
 
 legal_filename(Filename) ->
-    MaybeJson = filename:extension(Filename),
-    MaybeJson == ".json" orelse MaybeJson == <<".json">>.
+    case to_unicode_bin(Filename) of
+        <<"/", _/binary>> ->
+            %% never allow abs path
+            false;
```

Comment:
Not the scope here, but would probably have to check if the normalized path is in some allow-listed directories?

Reply:
You are right.
Will revert this change and add another task to follow up